nfs/mocks/nfsserver.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright © 2025 Dell Inc. or its subsidiaries. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mocks
+
```

Comment:
Can this mock be generated with mockgen similar to how we generate the Executor mock?

Reply:
Exploring at the moment. Will require a bit of adjustment to ensure things work as expected.

Please hold.

Reply:
Generate using `mockgen` and converted the unit tests to use that code.